tests/integration_cpp/basic.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2016-2020 Memgraph Ltd. [https://memgraph.com]
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "mgclient.hpp"
+
+template <typename T>
+T GetEnvOrDefault(const std::string &value_name, const T &default_value) {
+  const char *char_value = std::getenv(value_name.c_str());
+  if (!char_value)
+    return default_value;
+  T value;
+  std::stringstream env_value_stream(char_value);
+  env_value_stream >> value;
+  return value;
+}
+
+class MemgraphConnection : public ::testing::Test {
+protected:
+  virtual void SetUp() override {
+    client = mg::Client::Connect(
+        {GetEnvOrDefault<std::string>("MEMGRAPH_HOST", "127.0.0.1"),
+         GetEnvOrDefault<uint16_t>("MEMGRAPH_PORT", 7687), "", "",
+         GetEnvOrDefault<bool>("MEMGRAPH_SSLMODE", true), ""});
+  }
+
+  virtual void TearDown() override {
+    // Clean the result stream to unblock execute
+    std::optional<std::vector<mg::Value>> result;
+    while ((result = client->FetchOne()))
+      ;
```

Comment:
If there are any leftover rows in the stream at this point that should be an error condition because the test didn't consume its data correctly.
Change this to an `ASSERT_FALSE(client->FetchOne())`.

Reply:
What if the assert fails while we're still fetching rows? 
Then the execute will be blocked if I understood it correctly.

Reply:
If the `ASSERT` fails, the test will halt its execution, so it doesn't matter that the below execute will be blocked because it will never be executed.
The point of the `ASSERT` (instead of the whole current existing block) is to assert that there are no leftover rows in the client stream *after* the whole body of the test is done with its execution.

Reply:
But the rows we created will stay in the database which means that the test will fail again as it will contain extra rows that the test does not expect.

Reply:
That isn't an issue. You just have to have the cleanup section both in the `SetUp` and `TearDown` functions.